### ISSUE TITLE
Add query parameters argument to Router.prototype.url method

### ISSUE THREAD

**Issue by nschloe** [alexmingoia/koa-router#227]
koa-router's `url` method [supports name and parameters](https://github.com/alexmingoia/koa-router#module_koa-router--Router+url). In addition to this, I would like to construct a URL with query parameters, e.g.,

```
/test?q=abc&number=123
```

Is it possible to use the url method for it yet? Proper escaping would be nice too.

**Comment by mastilver** [comment]
It's already available using https://github.com/koajs/koa

Check out: https://github.com/koajs/koa/blob/9f80296fc49fa0c03db939e866215f3721fcbbc6/test/request/query.js#L23

```js
router.get('/test', function *(next) {
    var ctx = this;

    console.log(ctx.query.q);
    // => 'abc'
    console.log(ctx.query.number);
    // => 123
});
```

**Comment by nschloe** [comment]
Thanks for the reply! Do you think the same would be possible for the `url` method? Right now, I'm using it in the context of
```js
this.status = 303;
this.redirect(router.url('document', doc.document) + '?revision=' + doc._id);
```
The latter `?revision=...` part is the one that been bugging me.

**Comment by mastilver** [comment]
It's not handled

What do you think about:
```
router.url('document', doc.document, {
    query: {
        revision: doc._id
    }
})
```

I will try to send a PR tonight

**Comment by nschloe** [comment]
Just what I was looking for.

**Comment by wachunei** [comment]
Hi. Are there any plans on supporting this?

**Comment by jbielick** [comment]
@wachunei No plans that I've seen, but a PR could be opened with some work that matches what was discussed here—seems like there would be support for it.

**Comment by wachunei** [comment]
PR sent, check #396

**Comment by jbielick** [comment]
published at [`7.3.0`](https://www.npmjs.com/package/koa-router)
